### PR TITLE
Sbiswas/identify raster cell

### PR DESCRIPTION
**Pull request by sbiswas96**
Please review this sample that identifies the raster cell at a clicked point and outputs the attributes of the cell in a callout. The user can also click and drag around the area to get updates on the raster cells on the fly.

Sample design is [here](https://devtopia.esri.com/runtime/common-samples/tree/master/designs/IdentifyRasterCell).

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        m_mapView->calloutData()->setLocation(m_clickedPoint);
+        m_mapView->calloutData()->setDetail(calloutString);
+        m_mapView->calloutData()->setVisible(true);
+        m_calloutData = m_mapView->calloutData();
+        emit calloutDataChanged();
+      }
+    }
+  });
+
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](const QMouseEvent& e)
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.cpp:140]: do we want to identify on both `mouseClicked` and `mousePressedAndHeld`? Seems like we should use one or the other

File: ArcGISRuntimeSDKQt_CppSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            id: callout
+            calloutData: view.calloutData
+            accessoryButtonHidden: true
+            calloutContent: customComponent
+            leaderHeight: 30
+        }
+
+
+        Component {
+            id: customComponent
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml:39]: why do we need this component?

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            calloutData: parent.calloutData
+            autoAdjustWidth: false
+            calloutWidth: 300
+            accessoryButtonHidden: true
+            calloutContent: customComponent
+            leaderHeight: 30
+        }
+
+        Component {
+              id: customComponent
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml:48]: again, i'm not sure why we need this guy

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            }
+        }
+
+        onIdentifyLayerStatusChanged: {
+            if (identifyLayerStatus !== Enums.TaskStatusCompleted)
+                return;
+
+            for (let i = 0; i < identifyLayerResult.geoElements.length; i++) {
+                calloutText = "";
+                const attributes = identifyLayerResult.geoElements[i].attributes;
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml:114]: maybe get the `geoElement` once and cache it

File: ArcGISRuntimeSDKQt_CppSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml (hunk 2 above)
Comment (sbiswas96) [reply ArcGISRuntimeSDKQt_CppSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.qml:39]: this is added to be able to set the "calloutHeight" property, which is read-only so couldn't be set within the callout itself

**Pull request merged**